Web/src/components/games/PandaSequence/GameDisplay.tsx

```diff
@@ -57,19 +68,45 @@ const Score = styled.h3`
   padding-top: 6px;
 `;
 
+/**
+ * This component handles both display and input mode.
+ * Display Mode: Element in sequence is animated according to "displaying" prop
+ * Input Mode: Element tapped is animated based on "selected" state
+ */
 const GameDisplay: React.FC<IProps> = ({
-  mode, secondsLeft, score, active, handleInputEvent,
+  mode, secondsLeft, score, processInput, timestep, displaying,
 }) => {
-  const onTap = (index:number) => {
-    // TODO: Animate tapped component to confirm user input
-    handleInputEvent(index);
+  const [selected, setSelected] = useState(Array(5).fill(false));
+
+  const handleTouch = (index:number) => {
+    setSelected((oldSelected) => Object.assign([], oldSelected, { [index]: true }));
+  };
+
+  const handleTouchEnd = (index:number) => {
+    processInput(index);
+    setSelected(Object.assign([], selected, { [index]: false }));
   };
 
+
   let balloons;
   if (mode === PandaSequenceMode.DISPLAY) {
-    balloons = Array.from(Array(5).keys()).map((i) => <Balloon key={i} style={{ width: active === i ? '100px' : '70px' }} />);
+    balloons = Array.from(Array(5).keys()).map((index) => (
+      <DisplayBalloon
+        key={index}
+        duration={displaying === index ? timestep : 0}
+      />
+    ));
   } else {
-    balloons = Array.from(Array(5).keys()).map((i) => <Balloon key={i} onClick={() => onTap(i)} />);
+    balloons = Array.from(Array(5).keys()).map((index) => (
+      <InputBalloon
+        onTouchStart={() => handleTouch(index)}
```

Comment:
I think `ontouchstart` will also call `onmousedown` unless you call `preventDefault()`

Reply:
Sure thing let me see how to best handle this! I was indeed having problems with this. The following events are picked up:
**Mobile**
`onMouseDown`
`onTouchStart`
**Web**
`onMouseDown`
Right now I simply omitted the mouse events since our games are meant to be played on mobile, but I will look into using `preventDefault` to support both platforms.

Reply:
The idea basically is that onTouchStart gets invoked before onMouseDown (or the other way round, but basically, they are invoked in a certain order, not simultaneously). Once `preventDefault` is called on the event, this "bubbling" of event will be stopped, thus preventing `onMouseDown` to be invoked again after `onTouchStart`.

So, in the component you can do something like:

```typescript
onTouchStart={(e: React.SyntheticEvent) => handleTouch(e, index)}
onMouseDown={(e: React.SyntheticEvent) => handleTouch(e, index)}
```

and in `handleTouch`:

```typescript
const handleTouch = (e: React.SyntheticEvent, index: number) => {
  e.preventDefault();
  // ...
}
```

Reply:
Thanks so much for the suggestion!! I opened a can of worms about preventDefault being disabled by default due to [Chrome's use of passive events to improve scroll performance](https://developers.google.com/web/updates/2016/06/passive-event-listeners), and saw that [this is a problem for React because we don't add the event listeners ourselves](https://github.com/facebook/react/issues/8968). We could mount the events ourselves, but that would be terribly unelegant. I found the [solution](https://developers.google.com/web/updates/2017/01/scrolling-intervention#breakage_and_guidance) in setting the CSS property `touch-action: none`  on the input balloon to tell the browser we won't be scrolling. Crossing my fingers this would fix the issue once and for all.

Reply:
Seems to work :D